docs/guides/typescript.md

```diff
@@ -76,4 +76,55 @@ Remix has TypeScript type definitions built-in as well. The starter templates cr
 
 <docs-info>Note that the types referenced in `remix.env.d.ts` will depend on which environment you're running your app in. For example, there are different globals available in Cloudflare</docs-info>
 
+## Type Utilities
+
+Remix's built-in inference between `loader` function + `useLoaderData` generic and the `action` function + `useActionData` generic is quite good and you shouldn't often have to use these utilities. That said they can be quite handy in some cases which is why they are exported.
+
+### `UseDataFunctionReturn`
```

Comment:
`UseDataFunctionReturn` was renamed to `SerializedFrom` and can currently be imported from any one of the server runtime pkgs.